src/Nncase.Passes/Rules/Neutral/AddPostProcess.cs

```diff
@@ -44,16 +40,16 @@ protected override Task<IRModule> RunCoreAsync(IRModule module, RunPassContext o
 
         if (preProcess && modelLayout != outputLayout && outputLayout != string.Empty)
         {
-            var newOutput = outputLayout switch
+            Expr newOutput;
+            if (entry.Body is IR.Tuple tuple)
             {
-                "NHWC" when modelLayout == "NCHW" => Transpose(entry.Body, new[] { 0, 2, 3, 1 }),
-                "NCHW" when modelLayout == "NHWC" => Transpose(entry.Body, new[] { 0, 3, 1, 2 }),
-                _ => Transpose(
-                    entry.Body,
-                    Array.ConvertAll(
-                        outputLayout.Replace(" ", string.Empty, StringComparison.OrdinalIgnoreCase).Split(","),
-                        int.Parse)),
-            };
+                newOutput = new IR.Tuple(tuple.Fields.ToArray().Select((field, i) => AddTranspose(outputLayout, modelLayout, field)).ToArray());
+            }
+            else
+            {
+                newOutput = AddTranspose(outputLayout, modelLayout, entry.Body);
+            }
+
             var newEntry = entry.With(body: newOutput);
             module.Remove(entry);
             module.Add(newEntry);
```

Comment:
The `module.Add` can not update the entry index, it need to add `module.Entry = newEntry`.